spec/notification_center_spec.rb

```diff
@@ -0,0 +1,484 @@
+         .to eq(1)
+      end
+    end
+    
+    describe 'test add notification for multiple notification types' do
+      it 'should add, return notification ID and validate callbacks when a valid callback is added for each notification type ' do
+        Optimizely::NotificationCenter::NOTIFICATION_TYPES.values.each_with_index {|value, index|
+          notification_id = index+1
+          expect(notification_center.add_notification_listener(
+           value,
+           @callback_reference
+          )).to eq(notification_id)
+        }
+        notification_center.notifications.each do |key, _array|
+          notification_center.notifications[key].each do |notification|
+            expect(notification[:callback]).to eq @callback_reference
+          end
+        end
+      end
+      
+      it 'should add, return notification ID and validate callbacks when multiple valid callbacks are added for a single notification type' do
+        class CallBackSecond
+          def call(_args)
+            'Test multi listner.'
+          end
+        end
+  
+        @callback_second = CallBackSecond.new
+        @callback_reference_second = @callback_second.method(:call)
+        
+        notification_type = Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE]
+
+        expect(notification_center.add_notification_listener(
+          notification_type,
+         @callback_reference)
+        ).to eq(1)
+        
+        expect(notification_center.add_notification_listener(
+          notification_type,
+         @callback_reference_second)
+        ).to eq(2)
+        
+        expect(notification_center.notifications[notification_type][0][:callback]).to eq @callback_reference
+        expect(notification_center.notifications[notification_type][1][:callback]).to eq @callback_reference_second
+      end
+    end
+    
+    
+    describe 'test add notification that already added callback is not re-added' do
+      it 'should return -1 if callback already added' do
+        notification_center.add_notification_listener(
+         Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE],
+         @callback_reference
+        )
+        expect(notification_center.add_notification_listener(
+         Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE],
+         @callback_reference
+        )).to eq(-1)
+      end
+      
+      it 'should add same callback for a different notification type' do
+        expect(notification_center.add_notification_listener(
+         Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE],
+         @callback_reference
+        )).to eq(1)
+        
+        expect(notification_center.add_notification_listener(
+         Optimizely::NotificationCenter::NOTIFICATION_TYPES[:TRACK],
+         @callback_reference
+        )).to eq(2)
+      end
+    end
+    
+    describe '@error_handler' do
```

Comment:
Move this test case line 162 - 172 at the end of the file. For better organization